common/digest/sha1.c

```diff
@@ -0,0 +1,230 @@
+       (y)[2] = (unsigned char)(((x)>>8)&255); (y)[3] = (unsigned char)((x)&255); } while(0)
+
+#define STORE64H(x, y)                                                                     \
+do { (y)[0] = (unsigned char)(((x)>>56)&255); (y)[1] = (unsigned char)(((x)>>48)&255);     \
+     (y)[2] = (unsigned char)(((x)>>40)&255); (y)[3] = (unsigned char)(((x)>>32)&255);     \
+     (y)[4] = (unsigned char)(((x)>>24)&255); (y)[5] = (unsigned char)(((x)>>16)&255);     \
+     (y)[6] = (unsigned char)(((x)>>8)&255); (y)[7] = (unsigned char)((x)&255); } while(0)
+
+#define F0(x,y,z)  (z ^ (x & (y ^ z)))
+#define F1(x,y,z)  (x ^ y ^ z)
+#define F2(x,y,z)  ((x & y) | (z & (x | y)))
+#define F3(x,y,z)  (x ^ y ^ z)
+
+struct sha1_state {
+    ulong64 length;
+    ulong32 state[5], curlen;
+    unsigned char buf[64];
+};
+
+static int s_sha1_compress(struct sha1_state *md, const unsigned char *buf)
+{
+    ulong32 a,b,c,d,e,W[80],i;
+    ulong32 t;
+
+    /* copy the state into 512-bits into W[0..15] */
+    for (i = 0; i < 16; i++) {
+        LOAD32H(W[i], buf + (4*i));
+    }
+
+    /* copy state */
+    a = md->state[0];
+    b = md->state[1];
+    c = md->state[2];
+    d = md->state[3];
+    e = md->state[4];
+
+    /* expand it */
+    for (i = 16; i < 80; i++) {
+        W[i] = ROL(W[i-3] ^ W[i-8] ^ W[i-14] ^ W[i-16], 1);
+    }
+
+    /* compress */
+    /* round one */
+    #define FF0(a,b,c,d,e,i) e = (ROLc(a, 5) + F0(b,c,d) + e + W[i] + 0x5a827999UL); b = ROLc(b, 30);
+    #define FF1(a,b,c,d,e,i) e = (ROLc(a, 5) + F1(b,c,d) + e + W[i] + 0x6ed9eba1UL); b = ROLc(b, 30);
+    #define FF2(a,b,c,d,e,i) e = (ROLc(a, 5) + F2(b,c,d) + e + W[i] + 0x8f1bbcdcUL); b = ROLc(b, 30);
+    #define FF3(a,b,c,d,e,i) e = (ROLc(a, 5) + F3(b,c,d) + e + W[i] + 0xca62c1d6UL); b = ROLc(b, 30);
+
+    for (i = 0; i < 20; ) {
+       FF0(a,b,c,d,e,i++); t = e; e = d; d = c; c = b; b = a; a = t;
+    }
+    for (; i < 40; ) {
+       FF1(a,b,c,d,e,i++); t = e; e = d; d = c; c = b; b = a; a = t;
+    }
+    for (; i < 60; ) {
+       FF2(a,b,c,d,e,i++); t = e; e = d; d = c; c = b; b = a; a = t;
+    }
+    for (; i < 80; ) {
+       FF3(a,b,c,d,e,i++); t = e; e = d; d = c; c = b; b = a; a = t;
+    }
+
+    #undef FF0
+    #undef FF1
+    #undef FF2
+    #undef FF3
+
+    /* store */
+    md->state[0] = md->state[0] + a;
+    md->state[1] = md->state[1] + b;
+    md->state[2] = md->state[2] + c;
+    md->state[3] = md->state[3] + d;
+    md->state[4] = md->state[4] + e;
+
+    return 0;
+}
+
+static int sha1_init(struct sha1_state * md)
+{
+   if (md == NULL) return -1;
+   md->state[0] = 0x67452301UL;
+   md->state[1] = 0xefcdab89UL;
+   md->state[2] = 0x98badcfeUL;
+   md->state[3] = 0x10325476UL;
+   md->state[4] = 0xc3d2e1f0UL;
+   md->curlen = 0;
+   md->length = 0;
+   return 0;
+}
+
+static int sha1_process(struct sha1_state * md, const unsigned char *in, unsigned long inlen)
+{
+    unsigned long n;
+    int err;
+    if (md == NULL) return -1;
```

Comment:
These ifs are terrible, because the caller does not check for the return value. Please get rid of them everywhere and make init, process, and done return void.

Reply:
To the contrary, the caller checks those values sometimes, see `*_process`, and I do not want to remove them because it will make porting upcoming changes from LibTomCrypt harder. All those hash implementation files got a "minimal set of changes to make them a) build, b) work, c) standalone, if possible", nothing else.